src/simsopt/geo/biotsavart.py

```diff
@@ -1,27 +1,55 @@
 import numpy as np
-import simsgeopp as sgpp
+import simsoptpp as sopp
 from simsopt.geo.magneticfield import MagneticField
 
 
-class BiotSavart(MagneticField):
+class BiotSavart(sopp.BiotSavart, MagneticField):
+    r"""
+    Computes the MagneticField induced by a list of closed curves :math:`\Gamma_k` with electric currents :math:`I_k`.
+    The field is given by
+
+    .. math::
+
+        B(\mathbf{x}) = \frac{\mu_0}{4\pi} \sum_{k=1}^{n_\mathrm{coils}} I_k \int_0^1 \frac{(\Gamma_k(\phi)-\mathbf{x})\times \Gamma_k'(\phi)}{\|\Gamma_k(\phi)-\mathbf{x}\|^3} d\phi
+
+    where :math:`\mu_0=4\pi 10^{-7}` is the magnetic constant.
+
+    """
 
     def __init__(self, coils, coil_currents):
         assert len(coils) == len(coil_currents)
+        self.currents_optim = [sopp.Current(c) for c in coil_currents]
+        self.coils_optim = [sopp.Coil(curv, curr) for curv, curr in zip(coils, self.currents_optim)]
+        MagneticField.__init__(self)
+        sopp.BiotSavart.__init__(self, self.coils_optim)
```

Comment:
coils_optim is set in a derived class and then passed as an argument to base class. Kind of unusual. What is the logic behind this?

Reply:
A coil object essentially just consists of a curve together with a current object. Once the new graph structure is fully ready, that's how we want user to do coil optimisation. It will naturally allow for scenarios where several coils have the same current for example. The constructor to Biot Savart will then just take a list of coils.

However, since that's not ready yet, I don't want to break the old interface, and for now the user facing biot savart class takes separate curves and currents, and then internally creates the corresponding coil objects.